Doc/reference/simple_stmts.rst

```diff
@@ -802,7 +802,7 @@ The :keyword:`from` form uses a slightly more complex process:
 Examples::
 
    import foo                 # foo imported and bound locally
-   import foo.bar.baz         # foo, foo.bar, and foo.bar.baz imported, foo bound locally
+   import foo.bar.baz         # foo, foo.bar, and foo.bar.baz imported, foo.bar.baz bound locally
```

Comment:
Possibly we could explain this a little better, though. Do you think you might be able to come up with an alternative rewrite? :)

Reply:
With your explanation I understand what its intended meaning is. Although the line between `bound` and not is quite blurry, as with this 
```python
>>> import neo4j.spatial
>>> neo4j
<module 'neo4j' from '/home/stefano/virtualenvs/python-driver/lib/python3.10/site-packages/neo4j/__init__.py'>
>>> neo4j.spatial
<module 'neo4j.spatial' from '/home/stefano/virtualenvs/python-driver/lib/python3.10/site-packages/neo4j/spatial/__init__.py'>
>>> globals()
{'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <class '_frozen_importlib.BuiltinImporter'>, '__spec__': None, '__annotations__': {}, '__builtins__': <module 'builtins' (built-in)>, 'neo4j': <module 'neo4j' from '/home/stefano/virtualenvs/python-driver/lib/python3.10/site-packages/neo4j/__init__.py'>}
```
while only `neo4j` is bound locally, I can do `neo4j.spatial.<things>` as well. This is likely my own python ignorance, but what can I do with `neo4j` (bound locally) that I cannot do with `neo4j.spatial` (not bound)? Is there a difference in the behavior exposed to the end user?

Reply:
You're correct to say that the `neo4j.spatial` module is _bound_, but it is bound _as an attribute on the `neo4j module_ rather than _bound locally_ :)

> what can I do with `neo4j` (bound locally) that I cannot do with `neo4j.spatial` [...] ?

Not much. However, exactly which namespace a name is present in can often be an important detail!  It's not that uncommon to have code that iterates over a module's `globals` dictionary, or over a class's dictionary. This section of the docs isn't a tutorial aimed at beginners; it's technical documentation that aims to be as precise as possible, so it's important to include these details.

However, this section of the docs should -- like all sections! -- still aim to be as _clear_ as possible. So I'm still open to a rewrite here, if you can think of a clearer way of explaining this :-)

Reply:
Here's an another REPL example to illustrate what I mean by "bound as an attribute on the `neo4j` module: after `neo4j.spatial` has been imported, the `neo4j` module has a `spatial` attribute that did not exist before:

```pycon
Python 3.12.0 (tags/v3.12.0:0fb18b0, Oct  2 2023, 13:03:39) [MSC v.1935 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import importlib
>>> importlib.util
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'importlib' has no attribute 'util'
>>> importlib.__dict__['util']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'util'
>>> importlib
<module 'importlib' from 'C:\\Users\\alexw\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\importlib\\__init__.py'>
>>> import importlib.util
>>> importlib.util
<module 'importlib.util' (frozen)>
>>> importlib.__dict__['util']
<module 'importlib.util' (frozen)>
>>> importlib
<module 'importlib' from 'C:\\Users\\alexw\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\importlib\\__init__.py'>
```